packages/camera/example/lib/main.dart

```diff
@@ -137,25 +138,123 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
 
   /// Toggle recording audio
   Widget _toggleAudioWidget() {
+    return Row(
+      children: <Widget>[
+        const Text('Enable Audio:'),
+        Switch(
+          value: enableAudio,
+          onChanged: (bool value) {
+            enableAudio = value;
+            if (controller != null) {
+              onNewCameraSelected(controller.description);
+            }
+          },
+        ),
+      ],
+    );
+  }
+
+  /// Display the control bar with buttons to options like audio and flash.
+  Widget _optionsControlRowWidget() {
     return Padding(
-      padding: const EdgeInsets.only(left: 25),
+      padding: const EdgeInsets.symmetric(horizontal: 10),
       child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        mainAxisSize: MainAxisSize.max,
         children: <Widget>[
-          const Text('Enable Audio:'),
-          Switch(
-            value: enableAudio,
-            onChanged: (bool value) {
-              enableAudio = value;
-              if (controller != null) {
-                onNewCameraSelected(controller.description);
-              }
-            },
-          ),
+          _toggleAudioWidget(),
+          _flashButton(),
+          _torchButton(),
         ],
       ),
     );
   }
 
+  /// Flash Toggle Button
+  Widget _flashButton() {
```

Comment:
I think that you can improve existing code by checking 'flashmode' using 'switch' statement. For now for the most of cases you're setting icon twice, which is not an expected behaviour.